.devcontainer/Mojo.Dockerfile

```diff
@@ -0,0 +1,179 @@
+
+ENV PATH=${LLVM_VERSION:+/usr/lib/llvm-}${LLVM_VERSION}${LLVM_VERSION:+/bin:}$PATH
+
+RUN dpkgArch="$(dpkg --print-architecture)" \
+  ## Ensure that common CA certificates
+  ## and OpenSSL libraries are up to date
+  && apt-get update \
+  && apt-get -y install --no-install-recommends --only-upgrade \
+    ca-certificates \
+    openssl \
+## Install Modular
+  && apt-get -y install --no-install-recommends \
+    apt-transport-https \
+  && curl -s https://dl.modular.com/public/installer/setup.deb.sh | bash - \
+  && apt-get -y install --no-install-recommends \
+    modular \
+  && if [ "$MOJO_VERSION" = "nightly" ]; then \
+    ## Update Mojo nightly
+    modular update nightly/mojo; \
+  fi \
+  ## Clean up
+  && rm -rf /root/.cache \
+    /root/.ipython \
+    /root/.lldb \
+    /root/.local \
+## Install tools for buildig and testing the Mojo standard library
+  && if [ -n "$LLVM_VERSION" ]; then \
+    ## Install LLVM
+    apt-get -y install --no-install-recommends \
+      lsb-release \
+      software-properties-common; \
+    curl -sSLO https://apt.llvm.org/llvm.sh; \
+    chmod +x llvm.sh; \
+    ./llvm.sh "$LLVM_VERSION"; \
+    ## Clean up
+    rm -rf llvm.sh \
+      /root/.ssh \
+      /root/.wget-hsts; \
+  else \
+    ## Install FileCheck
+    apt-get -y install --no-install-recommends llvm-14-tools; \
+    cp -a /usr/lib/llvm-14/bin/FileCheck /usr/local/bin/FileCheck; \
+    ## Clean up
+    apt-get -y purge llvm-14-tools; \
+    apt-get -y autoremove; \
+  fi \
+  ## Install lit
+  && pip install --no-cache-dir lit \
+  ## Install pre-commit
+  && pip install --no-cache-dir pre-commit \
+  ## Install entr
+  && apt-get -y install --no-install-recommends entr \
+## Dev Container only
+  ## Install hadolint
+  && case "$dpkgArch" in \
+    amd64) tarArch="x86_64" ;; \
+    arm64) tarArch="arm64" ;; \
+    *) echo "error: Architecture $dpkgArch unsupported"; exit 1 ;; \
+  esac \
+  && apiResponse="$(curl -sSL \
+    https://api.github.com/repos/hadolint/hadolint/releases/latest)" \
+  && downloadUrl="$(echo "$apiResponse" | grep -e \
+    "browser_download_url.*Linux-$tarArch\"" | cut -d : -f 2,3 | tr -d \")" \
+  && echo "$downloadUrl" | xargs curl -sSLo /usr/local/bin/hadolint \
+  && chmod 755 /usr/local/bin/hadolint \
+  ## Create backup of root directory
+  && cp -a /root /var/backups \
+  ## Clean up
+  && rm -rf /tmp/* \
+  && rm -rf /var/lib/apt/lists/*
+
+FROM ${BUILD_ON_IMAGE}:${MOJO_VERSION} as files
+
+RUN mkdir /files
+
+COPY conf/shell /files
+COPY scripts /files
+
+  ## Ensure file modes are correct
+RUN find /files -type d -exec chmod 755 {} \; \
+  && find /files -type f -exec chmod 644 {} \; \
+  && find /files/usr/local/bin -type f -exec chmod 755 {} \; \
+  ## Copy skeleton files for root
+  && cp -r /files/etc/skel/. /files/root \
+  ## except .bashrc and .profile
+  && bash -c 'rm -rf /files/root/{.bashrc,.profile}' \
+  && chmod 700 /files/root
+
+FROM docker.io/koalaman/shellcheck:stable as sci
```

Comment:
**Question** Why do we need this since we don't currently run `shellcheck` in mojo CI?